build/source/driver/multi_driver.f90

```diff
@@ -239,79 +263,102 @@ real(dp),parameter        :: safety = 0.01_dp               ! safety factor to e
+  call get_vlines(fileUnit,dataLines,err,message); call handle_err(err,message)
+  close(fileUnit)
+  
+  ! get the number of snow and soil layers for each HRU
+  gru_struc(iGRU)%hruInfo(iHRU)%nSnow = 0 ! initialize the number of snow layers
+  gru_struc(iGRU)%hruInfo(iHRU)%nSoil = 0 ! initialize the number of soil layers
+  do iVar=1,size(dataLines)
+   ! split the line into an array of words
+   call split_line(dataLines(iVar),chardata,err,message); call handle_err(err,message)
+   ! check if the line contains initial conditions data (contains the word "snow" or "soil")
+   do iword=1,size(chardata)
+    if(chardata(iword)=='snow') gru_struc(iGRU)%hruInfo(iHRU)%nSnow = gru_struc(iGRU)%hruInfo(iHRU)%nSnow+1
+    if(chardata(iword)=='soil') gru_struc(iGRU)%hruInfo(iHRU)%nSoil = gru_struc(iGRU)%hruInfo(iHRU)%nSoil+1
+    if(chardata(iword)=='snow' .or. chardata(iword)=='soil') exit ! exit once read the layer type
+   end do
+   deallocate(chardata)
   end do
-  deallocate(chardata)
- end do
- deallocate(dataLines)
+  deallocate(dataLines)
 
+ end do  ! looping through HRUs
 end do  ! looping through HRUs
 
 ! **** END OF TEMPORARY CODE ***
 
-! allocate space for the time step and computeVegFlux flags (recycled for each HRU for subsequent calls to coupled_em)
-allocate(dt_init(nHRU),stat=err); call handle_err(err,'problem allocating space for dt_init')
-allocate(computeVegFlux(nHRU),stat=err); call handle_err(err,'problem allocating space for computeVegFlux')
-
-! initialize data structures
-call initStruct(&
-                ! input: model control
-                nHRU,       &    ! number of HRUs
-                nSnow,      &    ! number of snow layers in each HRU
-                nSoil,      &    ! number of soil layers in each HRU
-                ! input: data structures
-                timeStruct, &    ! model time data
-                forcStruct, &    ! model forcing data
-                attrStruct, &    ! local attributes for each HRU
-                typeStruct, &    ! local classification of soil veg etc. for each HRU
-                mparStruct, &    ! model parameters
-                indxStruct, &    ! model indices
-                progStruct, &    ! model prognostic (state) variables
-                diagStruct, &    ! model diagnostic variables
-                fluxStruct, &    ! model fluxes
-                derivStruct,&    ! model derivatives
-                bparStruct, &    ! basin-average parameters
-                bvarStruct, &    ! basin-average variables
-                ! output: error control
-                err,message)   ; call handle_err(err,message)
+! *****************************************************************************
+! (3b) allocate space for data structures
+! *****************************************************************************
+
+! loop through data structures
+do iStruct=1,size(structInfo)
```

Comment:
Why the do loop and select case here? Just have the list of calls to allocGlobal.